designs/CentrallyManagingNuGetPackageVersions-VSManagePackages.md

```diff
@@ -0,0 +1,108 @@
+
+There would also be a potential issue when a single user gesture causes a package to be installed into multiple projects, particularly when the `Directory.Packages.props` file does not already contain a `PackageVersion` for the package. If NuGet calls APIs to modify all the projects in parallel, there might be a race condition with trying to update `Directory.Packages.props`.
+
+* API aware of centrally managed package versions
+
+Presumably this would mean there's a separate API to add a `PackageVersion` to the `Directory.Packages.props` file. But since this file is shared across multiple projects, would the API be independent from the projects, or should NuGet pick one project and call the API on that project?
+
+Should new APIs be added to the project systems considering the version attribute must not be written? Or NuGet could call the API with a null version. Or NuGet passes the version and the project system must know if the project uses centrally managed package versions or not.
+
+### Removing a package from a project
+
+Currently NuGet calls APIs on the project system to remove the `PackageReference` for that package given a package id. The current API is sufficient and the same behavior should continue. The `Directory.Packages.props` file should not be modified to remove the `PackageVersion`, as other projects may still be using it.
+
+### Change the version of a package in `Directory.Packages.props`
+
+This will affect, at minimum, all the loaded projects in the solution that have a `PackageReference` to the package whose version is changed. For SDK-style projects, it appears this should work already, by watching the file for changes and then sending NuGet a project nomination with the new information. Currently non-SDK style projects need to be unloaded and reloaded to take into account changes to imported MSBuild files.
+
+Given this single user gesture will often affect multiple projects, should the API to modify the `Directory.Packages.props` file be on a project system interface, or a new interface that's decoupled from individual projects?
+
+Once the file is modified, should NuGet notify projects that it's changed, or should they detect it themselves?
+
+In particular, given that SDK style projects currently use file watchers to detect changes to imported files and automatically reload, but non-SDK style projects do not and continue to use outdated information until the project is reloaded, how can we avoid a bad user experience when a customer uses Package Manager UI/Console to change package versions?
+
+## References
+
+* https://github.com/NuGet/Home/wiki/Centrally-managing-NuGet-package-versions
+* https://github.com/microsoft/MSBuildSdks/tree/master/src/CentralPackageVersions
+
+## Appendix 1 : Current project system API usage by NuGet
```

Comment:
You might already be aware, but I tried to capture a similar summary in https://github.com/NuGet/Home/blob/d5c73c912fcd44b0df2de457da78909ff2ca6b78/designs/NuGetExtensibility/PackageInstaller.md#csproj-packagereference-scenarios.